File: js/problem4.js
//solution: 906609
print(eulerProblem4());

function eulerProblem4() {
    var max = 0;
    for(var i = 100; i < 1000; ++i) {
        for(var j = 100; j < 1000; ++j) {
            var p = i*j;
            if(isPalindrome(p.toString()) && p > max) {
                max = p;
            }
        }
    }
    return max;
}

function isPalindrome(str) {
    var rev = strReverse(str);
    return str == rev;
}

function strReverse(str) {
    var ret = "";
    for(var i = str.length - 1; i >= 0; --i) {
        ret += str.charAt(i);
    }
    return ret;
}

File: python/problem4.py
#!/usr/bin/python
#solution: 906609

def is_palindrome(str):
    for i in range(len(str) / 2):
        if str[i] != str[len(str) - i - 1]:
            return False
    return True

def euler_problem_4():
    max = 0
    for i in range(100, 999, 1):
        for j in range(100, 999, 1):
            p = i * j
            if is_palindrome(str(p)) and max < p:
                max = p
    return max 

print euler_problem_4()
